test/helper/example_log_lines_test_data.ts

```diff
@@ -0,0 +1,766 @@
+// Unit test data mapped to the logdef examples in /util/example_log_lines.ts
+
+import { RepeatingFieldsTypes } from '../../resources/netlog_defs';
+import NetRegexes from '../../resources/netregexes';
+import Regexes from '../../resources/regexes';
+import { NetFields } from '../../types/net_fields';
+import { RepeatingFieldsExtract } from '../../types/net_props';
+import { CactbotBaseRegExp } from '../../types/net_trigger';
+import { ExampleLineName, ExampleRegex } from '../../util/example_log_lines';
+
+import { RegexUtilParams } from './regex_util';
+
+type RegexFunc<T extends ExampleLineName> = (params?: RegexUtilParams) => CactbotBaseRegExp<T>;
+
+export type ExampleLineNameWithRepeating = Extract<ExampleLineName, RepeatingFieldsTypes>;
+
+type NetFieldsStrings<T extends ExampleLineName> = { [field in keyof NetFields[T]]?: string };
+export type TestFields<T extends ExampleLineName> = T extends ExampleLineNameWithRepeating
+  ? NetFieldsStrings<T> & RepeatingFieldsExtract<T>
+  : NetFieldsStrings<T>;
+
+export type UnitTest<T extends ExampleLineName> = {
+  // the index of the example in 'en' (in /util/example_log_lines.ts) to test
+  indexToTest: number;
+  // override the regex to use for this test (as a func to support capture test)
+  regexOverride?: Partial<ExampleRegex<RegexFunc<T>>>;
+  expectedValues: TestFields<T>;
+};
+
+type LineTests = {
+  [T in ExampleLineName]: UnitTest<T> | UnitTest<T>[];
+};
+
+const lineTests: LineTests = {
+  GameLog: [
+    { // test base GameLog regex
+      indexToTest: 4,
```

Comment:
> You could potentially reference the exact line here instead of using an index, now that they're decoupled? Same thing in #138 once the tests and examples are decoupled there as well.

I think that's a good change.  Let me know your thoughts on how you want to handle merging #138/#139 (see comment on #138), and I'll figure out whether to make that change here or there.